### PR TITLE
[ws-deployment] Add gitpod installation step

### PR DESCRIPTION
**Pull request by princerachit**
## Description

Implement step to install gitpod. It invokes a script `install-gitpod.sh` which installs gitpod on the cluster.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/6332

## How to test

Tested it here with other PR changes: https://werft.gitpod-io-dev.com/job/ops-workspace-deploy-workspace-prs-install-gitpod.29/raw

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **princerachit** after the PR has been reviewed.

Associated issue: *#6332*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/workspace/deployment/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/workspace/deployment/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: operations/workspace/deployment/pkg/step/installgitpod.go [hunk 1]
```diff
@@ -17,18 +21,39 @@ const (
...
+	credFileEnvVar := fmt.Sprintf("GOOGLE_APPLICATION_CREDENTIALS=%s", context.Project.GCPSACredFile)
+	if _, err := os.Stat(context.Project.GCPSACredFile); errors.Is(err, os.ErrNotExist) {
+		// reset this to empty string so that we can fallback to default
+		// gcloud context. This is useful in local development and execution
+		// scenarios
+		credFileEnvVar = ""
+	}
+
+	// add versions manifest
+	valuesFiles := append(cluster.ValuesFiles, context.Gitpod.VersionsManifestFilePath)
```
Comment (princerachit) [inline operations/workspace/deployment/pkg/step/installgitpod.go:38]: Passing this as a values file did not work. As the content should be under the key, `gitpod`.

File: operations/workspace/deployment/pkg/orchestrate/deploy.go [hunk 2]
```diff
@@ -35,11 +35,15 @@ func Deploy(context *common.Context, clusters []*common.WorkspaceCluster) error
 	return nil
 }
 
-// TODO(prs): Add implementation once we have scripts in ops repo tested and
-// install step implemented
 func installGitpod(context *common.Context, cluster *common.WorkspaceCluster) error {
-	log.Log.Infof("received request to install gitpod on cluster: %s cannot be processed. Implementation pending", cluster.Name)
-	return nil
+	err := step.InstallGitpod(context, cluster)
```
Comment (MrSimonEmms) [inline operations/workspace/deployment/pkg/orchestrate/deploy.go:39]: Do line 39 and 40 want to swapped around?

Comment (princerachit) [reply operations/workspace/deployment/pkg/orchestrate/deploy.go:39]: good catch

Comment (princerachit) [reply operations/workspace/deployment/pkg/orchestrate/deploy.go]: Fixed

**Review by MrSimonEmms — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d94b1432fc4719d5f8f590d7187d0f2dc32e952d</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6619#pullrequestreview-802261049" title="Approved">MrSimonEmms</a>*

Associated issue: *#6332*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[operations/workspace/deployment/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/workspace/deployment/OWNERS)~~ [MrSimonEmms]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**